prisma-fmt/src/text_document_completion.rs

```diff
@@ -81,6 +97,41 @@ fn push_ast_completions(
                 });
             }
         }
+        ast::SchemaPosition::Model(
+            model_id,
+            ast::ModelPosition::Field(field_id, ast::FieldPosition::Attribute(attr_name, _, None)),
+        ) if datasource
+            .map(|ds| attr_name.starts_with(&ds.name) && attr_name.contains('.'))
+            .unwrap_or(false) =>
+        {
+            let field = &db.ast()[model_id][field_id];
+            let field_type = if let ast::FieldType::Supported(name) = &field.field_type {
+                name.name.as_str()
+            } else {
+                return;
+            };
+            let constructors = connector
+                .available_native_type_constructors()
+                .iter()
+                // Only the constructors matching the field's Prisma type.
+                .filter(|con| con.prisma_types.iter().any(|pt| pt.as_str() == field_type));
+            for constructor in constructors {
+                let name = constructor.name.to_owned();
+                let args = if constructor.number_of_args + constructor.number_of_optional_args == 0 {
+                    ""
+                } else {
+                    "($0)"
+                };
+
+                completion_list.items.push(CompletionItem {
+                    label: name.to_owned(),
+                    kind: Some(CompletionItemKind::CONSTRUCTOR),
+                    insert_text: Some(format!("{name}{args}")),
+                    insert_text_format: Some(InsertTextFormat::SNIPPET),
```

Comment:
Probably doesn't need to be a snippet here, plain text should be enough? (no need to put cursor inside `()`)

Though technically, we could do it for `@db.VarChar()` but then the completion needs to include `$0` like `@db.VarChar($0)`

Reply:
>Though technically, we could do it for @db.VarChar() but then the completion needs to include $0 like @db.VarChar($0)

This is what I tried to do (see line 123), but I could be wrong. This needs to be tested end-to-end by getting it into an editor before we merge.

Reply:
I missed that 😅 That should work then 👍🏼 
But we should have a setup where we could easily test this end to end at least in dev.

Reply:
Yes, I think Julius worked on that a bit. I'll try to see how easy I can make it as soon as I have some time.